op-node/rollup/derive/engine_controller.go

```diff
@@ -298,6 +322,64 @@ func (e *EngineController) InsertUnsafePayload(ctx context.Context, payload *eth
 	return nil
 }
 
+// tryBackupUnsafeReorg attempts to reorg(restore) unsafe head to backupUnsafeHead.
+// If succeeds, update current forkchoice state to the rollup node.
```

Comment:
It would be really nice to have metrics in here

Reply:
`TryBackupUnsafeReorg` method may call helper methods like `SetBackupUnsafeL2Head`, `SetUnsafeHead`. Inside this methods, metrics are implemented.